ihmc-high-level-behaviors/src/main/java/us/ihmc/PerceptionAndAutonomyProcess.java

```diff
@@ -567,6 +594,51 @@ public void updatePlanarRegions()
       }
    }
 
+   public void updateHeightMap()
+   {
+      if (realsenseDepthImage != null && realsenseDepthImage.isAvailable() && heightMapDemandNode.isDemanded())
+      {
+         RawImage latestRealsenseDepthImage = realsenseDepthImage.get();
+
+         if (rapidHeightMapExtractor == null)
+         {
+            rapidHeightMapExtractor = new RapidHeightMapExtractor(openCLManager, null); // TODO: Fix up for resetting to feet Z
+            rapidHeightMapExtractor.setDepthIntrinsics(latestRealsenseDepthImage.getIntrinsicsCopy());
+            rapidHeightMapExtractor.create(realsenseDepthImage, 1); // TODO: Can't use same image for whole run?
```

Comment:
@PotatoPeeler3000 The height map extractor wants to use the same depth image for the whole time, but I think @TomaszTB has it so there's a newly allocated RawImage with the data each frame, so we'll need to change that.

Reply:
The `RapidHeightMapExtractor` takes a `BytedecoImage` not a `RawImage`. Could it be easier to keep a `BytedecoImage` object as a field and simply update its contents each time we get a new `RawImage`?

Reply:
Why do we need to change that? I'm unsure of the problems that will cause

Reply:
Just create a BytedecoImage each update to keep it more self contained. I wouldn't want that as a field. This is what I did for planar regions.

Reply:
@ds58 Can't do that because the API is different than planar regions. I have to store as a field and copy into it each tick